tests/system/test_system.py

```diff
@@ -536,6 +536,13 @@ def query_docs(client):
         operation()
 
 
+@pytest.fixture
```

Comment:
looks good. can we add some tests running this with a transaction?

Reply:
I've added a new test that uses transaction in both test_system and test_system_async.